deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -656,6 +661,37 @@ def clear_fpgas(self) -> None:
                 with StreamLogger('stdout'), StreamLogger('stderr'):
                     run(f"xbutil reset -d {card_bdf} --force")
 
+    def localize_xclbin(self, slotno: int) -> None:
+        """ download xclbin URI to remote node. """
+        assert slotno < len(self.parent_node.sim_slots)
+        serv = self.parent_node.sim_slots[slotno]
+        hwcfg = serv.get_resolved_server_hardware_config()
+        if re.match(_RFC_3986_PATTERN, hwcfg.xclbin):
+            remote_home_dir = self.parent_node.get_sim_dir()
+            remote_sim_dir = """{}/sim_slot_{}/""".format(remote_home_dir, slotno)
+            hwcfg.local_xclbin = './local.xclbin'
+
+            def get_xclbin(xclbin_uri:str, xclbin_lpath:PathLike) -> None:
+                # TODO instead of blowing up if the file exists, consider using fsspec
+                # filecache https://filesystem-spec.readthedocs.io/en/latest/features.html#caching-files-locally
+                # so that multiple slots using the same xclbin only grab it once and
+                # we only download it if it has changed at the source.
+                # HOWEVER, 'filecache' isn't thread/process safe and I'm not sure whether
+                # this runs in @parallel for fabric
+                lpath = Path(xclbin_lpath)
+                assert not lpath.exists, f"{lpath.resolve(strict=False)} already exists, refusing to overwrite"
```

Comment:
This is going to fail often since people rerun infrasetup before runworkload right? For all other files, the current behavior of infrasetup is to silently clobber stuff that exists so we should probably do the same here no?

However, we could have the manager backup the directory on re-invocations of infrasetup or ensure the directory does not exist, because the behavior here is probably more desirable.

Reply:
Yeah.  You make a good point about this failing often.  The ideal thing to do would be using the FileCache so that fsspec will do a checksum and not re-download the file if you've already got it.  I'd just like to be confident that it is safe to use the FileCache (and that it remains safe).  

I really like the current behavior of infrasetup where the rsync is essentially a no-op if you need to rerun it.  I wouldn't want to change that (although I'm not sure that's what you were suggesting with your 'backup' idea).

Reply:
If you are concerned about this potentially breaking... I think grabbing a lock and releasing it should be fine (esp. since that wouldn't be the bottleneck).

Reply:
@abejgonzalez what kind of lock are you thinking?  `flock()`?  I typically stay away from `flock()` because it isn't implemented well in many of the NFSv3 environments I find myself having to work in.  I think I'll just change the code to overwrite the file for now and if we can revisit the FileCache in a future PR if needed.

Reply:
SGTM. I didn't have a lock in mind.

Reply:
Yeah, I'm going to punt on using the FileCache for now.  I made `localize_xclbin()` overwrite with a logged message in https://github.com/firesim/firesim/pull/1140/commits/9bd9bce1d0880909a3676b967d3f967dfd368bfd